File: frequencies.js
const A = 440; // Hz
const cLight = 299792458; // m/s

const notes = ["A", "Bb", "B", "C", "Db", "D", "Eb", "E", "F", "Gb", "G", "Ab"];

const visibleLightMaxNm = 750;

function note2frequency(noteName) {
  return A * 2 ** (notes.indexOf(noteName) / 12);
}

function frequency2wavelength(frequency) {
  return cLight / frequency;
}

function findVisibleLightOctave(frequency) {
  let octaves = 0;
  let wavelengthNm;
  while (
    (wavelengthNm = frequency2wavelength(frequency) * 1e9) > visibleLightMaxNm
  ) {
    octaves += 1;
    frequency *= 2;
  }
  return [wavelengthNm, octaves];
}

export function noteName2Wavelength(noteName) {
  const noteFrequency = note2frequency(noteName);
  const [wavelengthAsLight, numOctaves] = findVisibleLightOctave(noteFrequency);
  const wavelengthAsLightRounded = wavelengthAsLight.toFixed(2);
  return `${noteName}: ${wavelengthAsLightRounded} nm (${numOctaves} octaves up)`;
}

File: frequencies.py
A = 440  # Hz
C_LIGHT = 299_792_458  # m/s

notes = ["A", "Bb", "B", "C", "Db", "D", "Eb", "E", "F", "Gb", "G", "Ab"]

visible_light_min_nm = 380
visible_light_max_nm = 750


def note2frequency(note_name: str) -> float:
    return A * 2 ** (notes.index(note_name) / 12)


def frequency2wavelength(frequency: float) -> float:
    return C_LIGHT / frequency


def find_visible_light_octave(frequency: float) -> tuple[float, int]:
    octaves = 0
    while (wavelength_nm := frequency2wavelength(frequency) * 1e9) > visible_light_max_nm:
        octaves += 1
        frequency *= 2
    return wavelength_nm, octaves


if __name__ == "__main__":
    note_name = input(f"Which note? ({','.join(notes)})\n").title()
    note_frequency = note2frequency(note_name=note_name)
    wavelength_as_light, num_octaves = find_visible_light_octave(note_frequency)
    print(f"{note_name}: {wavelength_as_light:.2f} nm ({num_octaves} octaves up)")
